pkg/apis/v1/nodeclaim_validation.go

```
@@ -149,3 +153,30 @@ func ValidateRequirement(requirement NodeSelectorRequirementWithMinValues) error
 	}
 	return errs
 }
+
+// ValidateKnownValues checks if the requirement has well known values and validates that at least one of the values is valid.
+// It returns error if all values are invalid, which will prevent the nodepool from going ready, otherwise it fails open.
+func validateKnownValues(requirement NodeSelectorRequirementWithMinValues) error {
+	// If the key doesn't have well-known values or the operator is not In, nothing to validate
+	if !WellKnownLabels.Has(requirement.Key) || requirement.Operator != v1.NodeSelectorOpIn {
+		return nil
+	}
+
+	// If the key doesn't have well-known values defined, nothing to validate
+	knownValues, exists := WellKnownValuesForRequirements[requirement.Key]
+	if !exists {
+		return nil
+	}
+
+	rejectedValues := lo.Filter(requirement.Values, func(val string, _ int) bool {
+		return !knownValues.Has(val)
+	})
+
+	// If there are no valid values, set an error to prevent the nodepool from going ready
+	if len(rejectedValues) != 0 {
```

Why not fail if any values are invalid? I'd rather fail loudly than fail silently